central/image/service/service_impl.go

```diff
@@ -207,37 +208,71 @@ func (s *serviceImpl) ScanImageInternal(ctx context.Context, request *v1.ScanIma
 
 	defer s.internalScanSemaphore.Release(1)
 
-	imgID := request.GetImage().GetId()
+	var (
+		img       *storage.Image
+		fetchOpt  enricher.FetchOption
+		imgExists bool
+	)
 
+	imgID := request.GetImage().GetId()
 	// Always pull the image from the store if the ID != "". Central will manage the reprocessing over the images.
 	if imgID != "" {
 		existingImg, exists, err := s.datastore.GetImage(ctx, imgID)
 		if err != nil {
 			return nil, err
 		}
-		// If the scan exists, return the scan.
-		// Otherwise, run the enrichment pipeline.
+
+		// If the image exists, and the request's image name exists within the images name, we return the image which
+		// includes the scan information (which includes signature data).
+		// Otherwise, we run the enrichment pipeline using the existing image with the requests image being added to it.
 		if exists {
```

Comment:
I think this `if` is not covered by https://github.com/stackrox/stackrox/pull/3950, right? Maybe it makes sense to introduce a test here